ui/src/app/shared/header/header.component.ts

```diff
@@ -0,0 +1,148 @@
+import {Component, NgZone, OnInit} from '@angular/core';
+import {MdChipInputEvent} from '@angular/material';
+import {ENTER} from '@angular/cdk/keycodes';
+import {FormControl} from "@angular/forms";
+import {Observable} from "rxjs/Observable";
+
+import {URLSearchParamsUtils} from "../url-search-params.utils";
+import {ActivatedRoute, Router} from "@angular/router";
+import {JobStatus} from "../model/JobStatus";
+import {QueryJobsRequest} from "../model/QueryJobsRequest";
+import {environment} from "../../../environments/environment";
+import {QueryFields} from "../common";
+
+@Component({
+  selector: 'jm-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css'],
+})
+export class HeaderComponent implements OnInit {
+  separatorKeysCodes = [ENTER];
+  control: FormControl = new FormControl();
+  options: string[] = [];
+  currentChipKey: string = "";
+  currentChipValue: string = "";
+  inputValue: string = "";
+
+  chips: Map<string, string>;
+
+  filteredOptions: Observable<string[]>;
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    private readonly router: Router,
+    private zone: NgZone,
+  ) {
+    route.queryParams.subscribe(
+      params => this.refreshChips(params['q']))
+  }
+
+  ngOnInit(): void {
+    if (this.route.snapshot.queryParams['q']) {
+      this.chips = URLSearchParamsUtils.getChips(this.route.snapshot.queryParams['q']);
+    }
+    this.filterOptions();
+    this.options = URLSearchParamsUtils.getQueryFields();
+    if (environment.additionalColumns) {
+      this.options = this.options.concat(environment.additionalColumns);
+    }
+  }
+
+  addChip(value: string): void {
+    if ((value || '').trim()) {
+      if (value.split(': ').length == 2) {
```

Comment:
I think if the user types `Statuses:Failed` we should auto translate that to `Statuses: Failed` so that it still works. Also maybe we should display some error for chips which do not parse to a queryable. Right now you can enter anything and there's no indication that it's invalid.

Reply:
I don't really have any good suggestions, but I think that makes sense to have a UX discussion around it before implementing anything.